### PR TITLE
feat: add types and open video posts

### PR DESCRIPTION
**Pull request by sasicodes**
## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [ ] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [ ] I checked that files in the `assets` folder are used by the extension itself
- [ ] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by sasicodes** [comment]
cc @peduarte

**Comment by peduarte** [comment]
ohh nice nice! i have some other un-pushed changes where I started adding types, but this is very helpful! thank youuuuu

File: extensions/lenscast/src/profiles.tsx [hunk 1]
```diff
@@ -84,6 +64,15 @@ function Profile({ profileId, bio, name, handle, picture, stats }: ProfileData)
...
   const avatar = normalizeUrl(picture?.original?.url);
 
+  const getAppUrl = (post: Post) => {
+    switch (post.metadata.mainContentFocus) {
+      case PublicationMainFocus.Video:
+        return `https://lenstube.xyz/watch/${post.id}`;
+      default:
+        return `https://lenster.xyz/posts/${post.id}`;
+    }
+  };
```
Comment (peduarte) [inline extensions/lenscast/src/profiles.tsx:74]: did not know about `mainContentFocus`, thanks!

Comment (sasicodes) [reply extensions/lenscast/src/profiles.tsx:74]: @peduarte `mainContentFocus` is later added in metadata v2. It is a way of adding a publication category like image, video or audio, etc.

By providing categories, it is easy for clients to filter all publications containing specific content, that is what I'm doing in Lenstube (VIDEO).

More - https://docs.lens.xyz/docs/metadata-standards#maincontentfocus---required

**Pull request merged**